File: src/main/java/com/martmists/beeresourceful/libcd/CustomNectar.java
/*
 * Decompiled with CFR 0.148.
 *
 * Could not load the following classes:
 *  io.github.alloffabric.beeproductive.api.HoneyFlavor
 *  kotlin.Metadata
 *  kotlin.jvm.JvmOverloads
 *  kotlin.jvm.internal.Intrinsics
 *  net.minecraft.item.Item
 *  net.minecraft.item.ItemConvertible
 *  net.minecraft.item.ItemStack
 *  net.minecraft.item.Items
 *  org.jetbrains.annotations.NotNull
 */
package com.martmists.beeresourceful.libcd;

import com.martmists.beeresourceful.BeeResourceful;
import io.github.alloffabric.beeproductive.api.HoneyFlavor;
import net.minecraft.item.ItemStack;
import net.minecraft.item.Items;
import java.util.ArrayList;
import java.util.List;


public final class CustomNectar {

	private static final List<String> knownNames;
	public static final CustomNectar INSTANCE;


	public final List<String> getKnownNames() {
		return knownNames;
	}


	public final void addNectar(String name, ItemStack shearing, ItemStack bottling) {


		if (!knownNames.contains(name)) {
			knownNames.add(name);
			BeeResourceful.INSTANCE.registerFlavor(name, new HoneyFlavor(shearing, bottling));
		}
	}

	public static /* synthetic */ void addNectar$default(CustomNectar customNectar, String string, ItemStack itemStack, ItemStack itemStack2, int n, Object object) {
		if ((n & 2) != 0) {
			itemStack = new ItemStack(Items.HONEYCOMB, 3);
		}
		if ((n & 4) != 0) {
			itemStack2 = new ItemStack(Items.HONEY_BOTTLE, 1);
		}
		customNectar.addNectar(string, itemStack, itemStack2);
	}


	public final void addNectar(String name, ItemStack shearing) {
		CustomNectar.addNectar$default(this, name, shearing, null, 4, null);
	}


	public final void addNectar(String name) {
		CustomNectar.addNectar$default(this, name, null, null, 6, null);
	}

	static {
		INSTANCE = new CustomNectar();
		knownNames = new ArrayList();
	}
}

File: src/main/kotlin/com/martmists/beeresourceful/libcd/CustomNectar.kt
package com.martmists.beeresourceful.libcd

import com.martmists.beeresourceful.BeeResourceful
import io.github.alloffabric.beeproductive.api.HoneyFlavor
import io.github.cottonmc.libcd.api.tweaker.TweakerManager
import net.minecraft.item.ItemStack
import net.minecraft.item.Items

/*
* Custom LibCD Tweaker for adding custom nectar
* Note: Requires textures and langs to be added in a resourcepack
* */
object CustomNectar {
    val knownNames = mutableListOf<String>()

    @JvmOverloads
    fun addNectar(name: String, shearing: ItemStack = ItemStack(Items.HONEYCOMB, 3), bottling: ItemStack = ItemStack(Items.HONEY_BOTTLE, 1)) {
        if (name !in knownNames) {
            knownNames.add(name)
            BeeResourceful.registerFlavor(name, HoneyFlavor(shearing, bottling))
        }
    }

    fun __register() {
        TweakerManager.INSTANCE.addAssistant("beeresourceful.libcd.CustomNectar", this)
    }
}